overcast/runner/__init__.py

```diff
@@ -400,6 +443,24 @@ def provision_step(self, details):
         else:
             userdata = None
 
+        pending_nodes = set()
+        done          = set()
+
+        if details.get('retry-delay', False):
+            retry_delay = utils.parse_time(details['retry-delay'])
+        else:
+            retry_delay = 5
+
+        if details.get('timeout', False):
+            provision_timeout = utils.parse_time(details['timeout'])
+            deadline =  time.time() + provision_timeout
+        else:
+            provision_timeout = None
+            deadline = None
```

Comment:
I'm not sure this is where these settings belong. I think you globally either want to retry N times or not. I can't imagine a use case for retrying one provisioning step in a deployment sequence, but not another?